src/v0/destinations/marketo_bulk_upload/fetchJobStatus.js

```diff
@@ -97,38 +102,43 @@ const responseHandler = async (event, type) => {
   } else {
     throw new PlatformError('No csvHeader in metadata');
   }
+  const startTime = Date.now();
   const data = {};
   // create a map of job_id and data sent from server
   // {<jobId>: '<param-val1>,<param-val2>'}
-  input.forEach((i) => {
-    const response = headerArr.map((fieldName) => Object.values(i)[0][fieldName]).join(',');
-    data[i.metadata.job_id] = response;
-  });
-  const unsuccessfulJobIdsArr = [];
-  const reasons = {};
-  const startTime = Date.now();
-  // match marketo response data with received data from server
-  for (const element of responseArr) {
-    // split response by comma but ignore commas inside double quotes
-    const elemArr = element.split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
-    // ref :
-    // https://developers.marketo.com/rest-api/bulk-import/bulk-custom-object-import/#:~:text=Now%20we%E2%80%99ll%20make%20Get%20Import%20Custom%20Object%20Failures%20endpoint%20call%20to%20get%20additional%20failure%20detail%3A
-    const reasonMessage = elemArr.pop(); // get the column named "Import Failure Reason"
+input.forEach((i) => {
+  const response = headerArr.map((fieldName) => Object.values(i)[0][fieldName]).join(',');
+  data[i.metadata.job_id] = response;
+});
+  const fieldSchemaMapping = await getFieldSchema(accessToken, config.munchkinId)
+  const unsuccessfulJobInfo = checkEventStatusViaSchemaMatching(event, fieldSchemaMapping)
 
-    for (const [key, val] of Object.entries(data)) {
-      // joining the parameter values sent from marketo match it with received data from server
-      if (val === `${elemArr.join()}`) {
-        // add job keys if warning/failure
-        if (!unsuccessfulJobIdsArr.includes(key)) {
-          unsuccessfulJobIdsArr.push(key);
+if (Object.keys(unsuccessfulJobInfo).length === 0 ){
```

Comment:
Some jobs can be mismatched, some jobs can fail here when we check marketo response. Can't this be a scenario?

Reply:
Logically yes. Updated it as if `(Object.keys(unsuccessfulJobInfo).length === 0 || type === 'fail' )`

Reply:
still there will be some issue , will try to fix it

Reply:
solved